18F/identity-idp
# Support empty TroubleshootingOptions options to avoid rendering

Extracted from #6483

**Why**:

- For alignment to [the same behavior in the Ruby implementation](https://github.com/18F/identity-idp/blob/acc9b77e4d6aedb859dd983ab2c965d0714037f7/app/components/troubleshooting_options_component.rb#L12-L14)
- To support use-cases where conditional rendering of options could result in no options to be rendered, such as in LG-5918 where a barcode read failure has occurred and there is no associated service provider, since the only option to be presented in TroubleshootingOptions is to "Get help at SP"

**Implementation Notes:**

This required some revisions to how the in-person proofing pilot "New Features" options are displayed, since previously it assumed it would be preceded by another set of TroubleshootingOptions and would not need to render the top red bar. The changes here refactor this behavior to apply to any sibling TroubleshootingOptions, allowing for the pilot options to still render the bar if it is the only set of options present on the page.

**Testing instructions:**

There should be no visual regressions in the display of troubleshooting options including the in-person proofing pilot options:

1. Document capture step errors ([testing instructions](https://developers.login.gov/testing/#testing-identity-proofing))
2. Proofing error screens (e.g. http://localhost:3000/verify/session/errors/exception )

**Screenshots:**

Document Capture|Proofing Exception
---|---
![image](https://user-images.githubusercontent.com/1779930/173825012-21f8ca20-b655-4c5f-8086-68a473ec47c2.png)|![image](https://user-images.githubusercontent.com/1779930/173824727-7a3acb8e-5af2-439b-97dc-6717fda72217.png)
